packages/flutter/test/material/outlined_button_test.dart

```diff
@@ -1947,6 +1947,24 @@ void main() {
     expect(controller.value, <MaterialState>{MaterialState.disabled});
     expect(count, 1);
   });
+
+  testWidgets("OutlinedButton.styleFrom doesn't throw exception on passing only one cursor", (WidgetTester tester) async {
+    // This is a regression test for https://github.com/flutter/flutter/issues/118071.
+    await tester.pumpWidget(
+      Directionality(
```

Comment:
nit: maybe we don't need to specify the Directionality here?

Reply:
This is required for `_InkResponseStateWidget` 
```consple
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: null
  Actual: FlutterError:<No Directionality widget found.
          _InkResponseStateWidget widgets require a Directionality widget ancestor.
          The specific widget that could not find a Directionality ancestor was:
            _InkResponseStateWidget
          The ownership chain for the affected widget is: "_InkResponseStateWidget ← InkWell ←
DefaultTextStyle ← AnimatedDefaultTextStyle ← _InkFeatures-[GlobalKey#df43a ink renderer] ←
NotificationListener<LayoutChangedNotification> ← CustomPaint ← _ShapeBorderPaint ← PhysicalShape ←
_MaterialInterior ← ⋯"
          Typically, the Directionality widget is introduced by the MaterialApp or WidgetsApp widget
at the top of your application widget tree. It determines the ambient reading direction and is used,
for example, to determine how to lay out text, how to interpret "start" and "end" values, and to
resolve EdgeInsetsDirectional, AlignmentDirectional, and other *Directional objects.>

When the exception was thrown, this was the stack:
#4      main.<anonymous closure> (file:///Users/tahatesser/Code/flutter/packages/flutter/test/material/outlined_button_test.dart:1963:5)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///Users/tahatesser/Code/flutter/packages/flutter/test/material/outlined_button_test.dart line 1963
The test description was:
  OutlinedButton.styleFrom doesn't throw exception on passing only one cursor
════════════════════════════════════════════════════════════════════════════════════════════════════
Test failed. See exception logs above.
The test description was: OutlinedButton.styleFrom doesn't throw exception on passing only one cursor

✖ OutlinedButton.styleFrom doesn't throw exception on passing only one cursor

```

Reply:
@QuncCccccc 
Just Fixed `IconButton` and added a new test to check exceptions. Please take a look.

Reply:
Ah that‘s right. Thanks for the explanation!